From 75bc1a897ffb6eecbef23582b86e3c16f5220aea Mon Sep 17 00:00:00 2001
From: Masahiro Nakai <nakai@atmark-techno.com>
Date: Thu, 22 Nov 2018 17:46:06 +0900
Subject: [PATCH] temp: build test with latest zephyr and openthread

zephyr:
- https://github.com/zephyrproject-rtos/zephyr.git
- ccb57d766b005d59ae6fb5b8ccae809a9603a6cc

openthread:
- https://github.com/openthread/openthread.git
- 30222e8538d7aafe6596a932af7e11fefa740114
---
 include/net/openthread.h                             |  3 ++-
 subsys/net/l2/openthread/openthread.c                | 12 +++++++-----
 subsys/net/l2/openthread/openthread_utils.c          |  3 ++-
 subsys/net/lib/openthread/CMakeLists.txt             |  9 +++++++++
 subsys/net/lib/openthread/platform/alarm.c           |  2 +-
 subsys/net/lib/openthread/platform/diag.c            |  2 +-
 subsys/net/lib/openthread/platform/misc.c            |  2 +-
 subsys/net/lib/openthread/platform/platform-zephyr.h |  3 ++-
 subsys/net/lib/openthread/platform/platform.c        |  2 +-
 subsys/net/lib/openthread/platform/radio.c           | 10 +++++-----
 subsys/net/lib/openthread/platform/random.c          |  2 +-
 subsys/net/lib/openthread/platform/shell.c           |  2 +-
 12 files changed, 33 insertions(+), 19 deletions(-)

diff --git a/include/net/openthread.h b/include/net/openthread.h
index d2fa776842..691c6482de 100644
--- a/include/net/openthread.h
+++ b/include/net/openthread.h
@@ -11,7 +11,8 @@
 
 #include <net/net_if.h>
 
-#include <openthread/openthread.h>
+//#include <openthread/openthread.h>
+#include <openthread/instance.h>
 
 #ifdef __cplusplus
 extern "C" {
diff --git a/subsys/net/l2/openthread/openthread.c b/subsys/net/l2/openthread/openthread.c
index ae3c3a0605..f9f603faad 100644
--- a/subsys/net/l2/openthread/openthread.c
+++ b/subsys/net/l2/openthread/openthread.c
@@ -17,9 +17,11 @@
 #include <init.h>
 #include <misc/util.h>
 #include <misc/__assert.h>
-#include <openthread/openthread.h>
+//#include <openthread/openthread.h>
+#include <openthread/tasklet.h>
+#include <openthread/thread.h>
 #include <openthread/cli.h>
-#include <platform.h>
+//#include <platform.h>
 #include <platform-zephyr.h>
 
 #include "openthread_utils.h"
@@ -206,8 +208,8 @@ static enum net_verdict openthread_recv(struct net_if *iface,
 	/* Length inc. CRC. */
 	recv_frame.mLength = net_buf_frags_len(pkt->frags);
 	recv_frame.mChannel = platformRadioChannelGet(ot_context->instance);
-	recv_frame.mLqi = net_pkt_ieee802154_lqi(pkt);
-	recv_frame.mRssi = net_pkt_ieee802154_rssi(pkt);
+	recv_frame.mInfo.mRxInfo.mLqi = net_pkt_ieee802154_lqi(pkt);
+	recv_frame.mInfo.mRxInfo.mRssi = net_pkt_ieee802154_rssi(pkt);
 
 #if defined(CONFIG_OPENTHREAD_L2_DEBUG_DUMP_15_4)
 	net_hexdump_frags("Received 802.15.4 frame", pkt, true);
@@ -238,7 +240,7 @@ enum net_verdict openthread_send(struct net_if *iface, struct net_pkt *pkt)
 
 	NET_DBG("Sending Ip6 packet to ot stack");
 
-	message = otIp6NewMessage(ot_context->instance, true);
+	message = otIp6NewMessage(ot_context->instance, NULL);
 	if (message == NULL) {
 		goto exit;
 	}
diff --git a/subsys/net/l2/openthread/openthread_utils.c b/subsys/net/l2/openthread/openthread_utils.c
index a84897eac5..b667a47162 100644
--- a/subsys/net/l2/openthread/openthread_utils.c
+++ b/subsys/net/l2/openthread/openthread_utils.c
@@ -11,7 +11,8 @@
 #include <net/net_pkt.h>
 #include <net/openthread.h>
 
-#include <openthread/openthread.h>
+//#include <openthread/openthread.h>
+#include <openthread/ip6.h>
 
 #include "openthread_utils.h"
 
diff --git a/subsys/net/lib/openthread/CMakeLists.txt b/subsys/net/lib/openthread/CMakeLists.txt
index 6eae56c621..be68405100 100644
--- a/subsys/net/lib/openthread/CMakeLists.txt
+++ b/subsys/net/lib/openthread/CMakeLists.txt
@@ -121,6 +121,7 @@ set(configure_flags
   --enable-no-executables-hack
   --disable-docs
   --with-platform-info=zephyr
+#  --with-examples=nrf52840
 )
 
 # TODO: Simplify
@@ -128,6 +129,7 @@ set(ZEPHYR_MBEDTLS_CPPFLAGS "${ZEPHYR_MBEDTLS_CPPFLAGS} ")
 set(ZEPHYR_MBEDTLS_CPPFLAGS "-DMBEDTLS_CONFIG_FILE='\"mbedtls-config.h\"'")
 set(ZEPHYR_MBEDTLS_CPPFLAGS "${ZEPHYR_MBEDTLS_CPPFLAGS} -DMBEDTLS_USER_CONFIG_FILE='\"${CMAKE_CURRENT_SOURCE_DIR}/zephyr-mbedtls-config.h\"'")
 set(ZEPHYR_MBEDTLS_CPPFLAGS "${ZEPHYR_MBEDTLS_CPPFLAGS} -I${ot_SOURCE_DIR}/third_party/mbedtls")
+set(ZEPHYR_MBEDTLS_CPPFLAGS "${ZEPHYR_MBEDTLS_CPPFLAGS} -I${ot_SOURCE_DIR}/third_party/mbedtls/repo/include")
 set(ZEPHYR_MBEDTLS_CPPFLAGS "${ZEPHYR_MBEDTLS_CPPFLAGS} -I${ot_SOURCE_DIR}/third_party/mbedtls/repo.patched/include")
 set(ZEPHYR_MBEDTLS_CPPFLAGS "${ZEPHYR_MBEDTLS_CPPFLAGS} -I${ot_SOURCE_DIR}/third_party/mbedtls/repo.patched/include/mbedtls")
 
@@ -156,6 +158,7 @@ endif()
 if(CONFIG_OPENTHREAD_SHELL)
   list(APPEND configure_flags
     --enable-cli-app=all
+#     --enable-cli
     )
 endif()
 
@@ -165,6 +168,12 @@ if(CONFIG_OPENTHREAD_DIAG)
     )
 endif()
 
+if(CONFIG_OPENTHREAD_FTD)
+  list(APPEND configure_flags
+    --enable-ftd
+    )
+endif()
+
 list(APPEND cmd
   CONFIGURE_COMMAND ./configure ${configure_flags}
 )
diff --git a/subsys/net/lib/openthread/platform/alarm.c b/subsys/net/lib/openthread/platform/alarm.c
index a8b07baccb..20759e663b 100644
--- a/subsys/net/lib/openthread/platform/alarm.c
+++ b/subsys/net/lib/openthread/platform/alarm.c
@@ -15,7 +15,7 @@ LOG_MODULE_REGISTER(LOG_MODULE_NAME);
 #include <inttypes.h>
 
 #include <openthread/platform/alarm-milli.h>
-#include <platform.h>
+//#include <platform.h>
 
 #include <stdio.h>
 
diff --git a/subsys/net/lib/openthread/platform/diag.c b/subsys/net/lib/openthread/platform/diag.c
index d166f59416..3a6ad6c2fb 100644
--- a/subsys/net/lib/openthread/platform/diag.c
+++ b/subsys/net/lib/openthread/platform/diag.c
@@ -6,7 +6,7 @@
 
 #include <kernel.h>
 #include <openthread-config.h>
-#include <openthread/openthread.h>
+//#include <openthread/openthread.h>
 #include <openthread/platform/diag.h>
 
 #include "platform-zephyr.h"
diff --git a/subsys/net/lib/openthread/platform/misc.c b/subsys/net/lib/openthread/platform/misc.c
index eee103f70c..5a16c3893a 100644
--- a/subsys/net/lib/openthread/platform/misc.c
+++ b/subsys/net/lib/openthread/platform/misc.c
@@ -6,7 +6,7 @@
 
 #include <kernel.h>
 #include <misc/reboot.h>
-#include <openthread/types.h>
+//#include <openthread/types.h>
 #include <openthread/platform/misc.h>
 
 #include "platform-zephyr.h"
diff --git a/subsys/net/lib/openthread/platform/platform-zephyr.h b/subsys/net/lib/openthread/platform/platform-zephyr.h
index ea61a7f45c..2fb3e9a4be 100644
--- a/subsys/net/lib/openthread/platform/platform-zephyr.h
+++ b/subsys/net/lib/openthread/platform/platform-zephyr.h
@@ -15,7 +15,8 @@
 
 #include <stdint.h>
 
-#include <openthread/openthread.h>
+//#include <openthread/openthread.h>
+#include <openthread/instance.h>
 
 /**
  * This function initializes the alarm service used by OpenThread.
diff --git a/subsys/net/lib/openthread/platform/platform.c b/subsys/net/lib/openthread/platform/platform.c
index 47751c3318..7428eef8ea 100644
--- a/subsys/net/lib/openthread/platform/platform.c
+++ b/subsys/net/lib/openthread/platform/platform.c
@@ -11,7 +11,7 @@
  */
 
 #include <kernel.h>
-#include <openthread/openthread.h>
+//#include <openthread/openthread.h>
 #include <openthread/tasklet.h>
 
 #include "platform-zephyr.h"
diff --git a/subsys/net/lib/openthread/platform/radio.c b/subsys/net/lib/openthread/platform/radio.c
index 9463624f5a..e7874e3ed1 100644
--- a/subsys/net/lib/openthread/platform/radio.c
+++ b/subsys/net/lib/openthread/platform/radio.c
@@ -30,9 +30,9 @@ LOG_MODULE_REGISTER(LOG_MODULE_NAME);
 
 #include <openthread/platform/radio.h>
 #include <openthread/platform/diag.h>
-#include <platform.h>
+//#include <platform.h>
 
-#include <openthread/types.h>
+//#include <openthread/types.h>
 
 #include "platform-zephyr.h"
 
@@ -98,7 +98,7 @@ void platformRadioProcess(otInstance *aInstance)
 		radio_api->set_channel(radio_dev, sTransmitFrame.mChannel);
 		radio_api->set_txpower(radio_dev, tx_power);
 
-		if (sTransmitFrame.mIsCcaEnabled) {
+		if (sTransmitFrame.mInfo.mTxInfo.mCsmaCaEnabled) {
 			if (radio_api->cca(radio_dev) ||
 			    radio_api->tx(radio_dev, tx_pkt, tx_payload)) {
 				result = OT_ERROR_CHANNEL_ACCESS_FAILURE;
@@ -129,8 +129,8 @@ void platformRadioProcess(otInstance *aInstance)
 
 				ackPsdu[2] = sTransmitFrame.mPsdu[2];
 				ackFrame.mPsdu = ackPsdu;
-				ackFrame.mLqi = 80;
-				ackFrame.mRssi = -40;
+				ackFrame.mInfo.mRxInfo.mLqi = 80;
+				ackFrame.mInfo.mRxInfo.mRssi = -40;
 				ackFrame.mLength = 5;
 
 				otPlatRadioTxDone(aInstance, &sTransmitFrame,
diff --git a/subsys/net/lib/openthread/platform/random.c b/subsys/net/lib/openthread/platform/random.c
index 563f9c0b09..786646b2b4 100644
--- a/subsys/net/lib/openthread/platform/random.c
+++ b/subsys/net/lib/openthread/platform/random.c
@@ -6,7 +6,7 @@
 
 #include <kernel.h>
 #include <string.h>
-#include <openthread/types.h>
+//#include <openthread/types.h>
 #include <openthread/platform/random.h>
 
 #include "platform-zephyr.h"
diff --git a/subsys/net/lib/openthread/platform/shell.c b/subsys/net/lib/openthread/platform/shell.c
index b75a38281a..b4889ecb98 100644
--- a/subsys/net/lib/openthread/platform/shell.c
+++ b/subsys/net/lib/openthread/platform/shell.c
@@ -10,7 +10,7 @@
 #include <shell/shell.h>
 #include <shell/shell_uart.h>
 #include <openthread/cli.h>
-#include <platform.h>
+//#include <platform.h>
 
 #include "platform-zephyr.h"
 
-- 
2.11.0

